Release/include/pplx/threadpool.h

```diff
@@ -156,6 +173,9 @@ class threadpool
     std::vector<pthread_t> m_threads;
     boost::asio::io_service m_service;
     boost::asio::io_service::work m_work;
+
+    std::mutex m_connection_pool_map_mutex;
+    std::map<std::string, std::shared_ptr<web::http::client::details::asio_connection_pool>> m_connection_pool_map;
```

Comment:
A pool timer is added to the `asio_connection_pool`.  The pool is going to be cleared 60 seconds after the last `http_client` using it releases it.